core/src/main/java/com/amplifyframework/core/model/SchemaRegistry.java

```diff
@@ -104,19 +140,33 @@ public synchronized ModelSchema getModelSchemaForModelClass(@NonNull String clas
         return Immutable.of(modelSchemaMap);
     }
 
+    /**
+     * Retrieve the map of Model CustomTypeName => CustomTypeSchema.
+     * @return an immutable map of Model CustomTypeName => CustomTypeSchema
+     */
+    @NonNull
+    public Map<String, CustomTypeSchema> getCustomTypeSchemaMap() {
+        return Immutable.of(customTypeSchemaMap);
+    }
+
     /**
      * Creates a new instance.
      * @return A new instance
      */
     @NonNull
-    public static synchronized ModelSchemaRegistry instance() {
-        return new ModelSchemaRegistry();
+    public static synchronized SchemaRegistry instance() {
+        if (SchemaRegistry.instance == null) {
+            SchemaRegistry.instance = new SchemaRegistry();
+        }
+
+        return SchemaRegistry.instance;
```

Comment:
From [this use case](https://github.com/aws-amplify/amplify-android/blob/c241e5128f28bd4a69d501226458cbf86fa10c9b/aws-api-appsync/src/main/java/com/amplifyframework/api/aws/SelectionSet.java#L368-L369) of `SchemaRegistry`, it looks like `SchemaRegistry` should be a singleton. The original code didn't work and didn't break either as the original logic wasn't really depending on retrieving a valid model schema.